### PR TITLE
MGDOBR-1180 - Archive artifacts with junit test results

### PR DESCRIPTION
**Pull request by jiripetrlik**
[MGDOBR-1180](https://issues.redhat.com/browse/MGDOBR-1180)

Archive junit test results for E2E tests.

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: .github/workflows/dev-cluster-check.yml [hunk 1]
```diff
@@ -81,6 +81,13 @@ jobs:
         with:
           maven-command: clean verify -Pcucumber -Dparallel -Dgroups=${{ github.event.inputs.groups || ''}} -Devent-bridge.manager.url=$DEV_CLUSTER_MANAGER_URL -Dkeycloak.realm.url=https://sso.redhat.com/auth/realms/redhat-external -Dbridge.client.id=$DEV_SSO_CLIENT_ID -Dbridge.client.secret=$DEV_SSO_CLIENT_SECRET -Dmanaged.kafka.sso.auth-server-url=https://sso.redhat.com/auth/realms/redhat-external -Dtest.credentials.file=e2e-test-config.yaml -Dfail.cleanup.disable=${{ github.event.inputs.fail-cleanup-disable || 'false'}} -Dcleanup.disable=${{ github.event.inputs.cleanup-disable || 'false'}}
           working-directory: integration-tests
+      - name: Archive junit test results
+        uses: actions/upload-artifact@v3
+        with:
+          name: junit-test-results
+          path: |
+            integration-tests/target/failsafe-reports/TEST-*.xml
```
Comment (afalhambra) [inline .github/workflows/dev-cluster-check.yml:89]: just to double check. We only want to store integration test results and not unit tests results?

**Comment by afalhambra** [comment]
@jiripetrlik @sutaakar - just out of curiosity and not related to this PR.
I noticed we have two files named `stable-cluster-check.yml` and `stage-cluster-check.yml`.
Are they duplicated? shouldn't we have only one for stage environment?

**Review by sutaakar — APPROVED** [review]

**Comment by sutaakar** [comment]
@afalhambra no, we actually have both stable and stage environments.

File: .github/workflows/dev-cluster-check.yml (hunk 1 above)
Comment (jiripetrlik) [reply .github/workflows/dev-cluster-check.yml:89]: As far as I know we run only the integration tests in our cluster checks. Unit tests are run as part of Bridge - CI workflow.

Comment (afalhambra) [reply .github/workflows/dev-cluster-check.yml:89]: ok, thanks!

**Review by afalhambra — APPROVED** [review]

**Pull request merged**